CoreScriptsRoot/CoreScripts/Topbar.lua

```diff
@@ -743,8 +813,40 @@ local function Create3DSettingsIcon(topBarInstance)
 		end
 	end)
 
+
+	local function OnHamburger3DInput(actionName, state, inputObj)
+		if state ~= Enum.UserInputState.Begin then
+			return
+		end
+		menuItem:SetSettingsActive(true) --this button is only ever shown if the settings menu isn't already open, so it can only be true.
+	end
+
+	local eaterAction = game:GetService("HttpService"):GenerateGUID()
+	local function EnableHamburger3DInput(enable)
+		if enable then
+			ContextActionService:BindCoreAction("Hamburger3DInput", OnHamburger3DInput, false, Enum.KeyCode.Space, Enum.KeyCode.ButtonA)
+			ContextActionService:BindAction(eaterAction, function() end, false, Enum.KeyCode.Space, Enum.KeyCode.ButtonA)
```

Comment:
I think the only info I have is that I wish I'd left a comment. I'll try removing it to see if I can remind myself of why it's there.